charts/openshift-metering/templates/hive/hive-scripts-configmap.yaml

```diff
@@ -109,4 +109,35 @@ data:
     export HIVE_METASTORE_HADOOP_OPTS=" -Dhive.log.level=${HIVE_LOGLEVEL} "
     export HIVE_OPTS="${HIVE_OPTS} --hiveconf hive.root.logger=${HIVE_LOGLEVEL},console "
 
+    # Note(tflannag): Hive supports derby (default), mysql, postgres, and oracle databases,
+    # but we only allow users to configure a mysql or postgres database instance.
+    #
+    # The following attempts to determine the database that was configured by the metering
+    # operator by parsing the ConnectionURL parameter in hive-site.xml. We can make the same
+    # assumption that this file exists and that parameter has been set as we control that
+    # file as part of Hive's reconciliation process. We use faq to pull out that key's
+    # value and parse the URL's scheme. We can then use the value of that scheme in order
+    # to avoid determing what database needs to be passed to `schematool -dbType`.
+    #
+    # TODO(tflannag): we may want to use a more dedicated tool in order to parse a URL
+    # as there's currently no validation in place such that invalid ConnectionURL values
+    # being configured may result in unexpected values being returned. Going forward, we
+    # should likely use Ansible to parse this URL for us and setting that environment variable
+    # to avoid determining it using bash.
+    export SCHEMATOOL_BIN="${HIVE_HOME}/bin/schematool"
+    DATABASE_TYPE=$(faq -f xml -o json \
+        '.configuration.property[] | select (.name=="javax.jdo.option.ConnectionURL").value' \
```

Comment:
maybe this could just be `ConnectionURL` as then the class pathing info can change without it breaking. I think this class name is pretty locked in, so unlikely to change.

Reply:
Ended up going in a different direction by parsing the `spec.hive.spec.config.db.url` field in Ansible instead of attempting to parse an xml file using `faq` and hoping for the best, but agreed, that sounds reasonable to me.